lib/consul/async/json_endpoint.rb

```diff
@@ -167,9 +172,10 @@ def fetch
           conn: EventMachine::HttpRequest.new(conf.url, options)
         }
         cb = proc do
-          http = connection[:conn].get(build_request)
+          request_method = conf.request_method.to_sym
+          http = connection[:conn].send(request_method, build_request)
           http.callback do
-            if enforce_json_200 && http.response_header.status != 200 && http.response_header['Content-Type'] != 'application/json'
+            if enforce_json_200 && !(200..299).include?(http.response_header.status) && http.response_header['Content-Type'] != 'application/json'
```

Comment:
It seems like you're generating an array with a hundred entries and then testing each individual ones. This is called on each response, and there can be a lot of them when watching _everything_ why not just `http.response_header.status => 200 && http.response_header.status < 300` ?

Reply:
Digging ruby magic it seems this can just be `http.response_header.status.between?(200, 299)`

Reply:
`(200..299)` is a Range which has a specific implementation of include?
(https://github.com/ruby/ruby/blob/132b7eb104ed2f5d9966adb2f0edacb2406f40fb/range.c#L1425)